tsinfer/inference.py

```diff
@@ -2060,40 +2011,15 @@ def get_augmented_ancestors_tree_sequence(self, sample_indexes):
         return tables.tree_sequence()
 
 
-class ResultBuffer:
-    """
-    A wrapper for numpy arrays representing the results of a copying operations.
-    """
-
-    def __init__(self):
-        self.paths = {}
-        self.mutations = {}
-        self.lock = threading.Lock()
-        self.total_edges = 0
-
-    def clear(self):
-        """
-        Clears this result buffer.
-        """
-        self.paths.clear()
-        self.mutations.clear()
-        self.total_edges = 0
-
-    def set_path(self, node_id, left, right, parent):
-        with self.lock:
-            assert node_id not in self.paths
-            self.paths[node_id] = left, right, parent
-            self.total_edges += len(left)
-
-    def set_mutations(self, node_id, site, derived_state):
-        with self.lock:
-            self.mutations[node_id] = site, derived_state
-
-    def get_path(self, node_id):
-        return self.paths[node_id]
-
-    def get_mutations(self, node_id):
-        return self.mutations[node_id]
+@dataclasses.dataclass
+class Result:
```

Comment:
MatchResult or something similar might be more descriptive @benjeffery. See [here](https://github.com/jeromekelleher/tsinfer/blob/f9070616b58221ab02f01738108e0ce19043ad35/tsinfer/matching.py#L104) for a rough version of how I was imagining it to work later on. It might be worth adding the Path class now, so we go ``result.path.left`` rather than ``result.path_left``.

Reply:
Done